IdentityCore/src/cache/metadata/accountMetadata/MSIDAccountMetadata.m

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDAccountMetadata.h"
+#import "MSIDAccountIdentifier.h"
+#import "MSIDAuthority.h"
+#import "MSIDAuthorityFactory.h"
+#import "NSDictionary+MSIDExtensions.h"
+#import "MSIDAccountMetadataCacheKey.h"
+#import "MSIDRequestParameters.h"
+
+static const NSString *AccountMetadataURLMapKey = @"URLMap";
+
+@interface MSIDAccountMetadata()
+
+@property NSMutableDictionary *internalMap;
+
+@end
+
+@implementation MSIDAccountMetadata
+
+- (instancetype)initWithHomeAccountId:(NSString *)homeAccountId
+                             clientId:(NSString *)clientId
+
+{
+    if (!homeAccountId || !clientId) return nil;
+    
+    self = [super init];
+    if (self)
+    {
+        _homeAccountId = homeAccountId;
+        _clientId = clientId;
+        _internalMap = [NSMutableDictionary new];
+        _signInState = MSIDAccountMetadataStateSignedIn;
```

Comment:
just curious, why is signed in the default state?

Reply:
Because the major case of creating MSIDAccountMetadata is when we sign in a user. 
Also, when we want to set sign out state, most likely we won't forget to flip the state to signed-out :)